Sources/Rendering/WebGPU/Renderer/index.js

```diff
@@ -301,18 +356,77 @@ function vtkWebGPURenderer(publicAPI, model) {
     }
 
     const device = model._parent.getDevice();
+    // Normal Solid Color
     if (!model.clearFSQ) {
       model.clearFSQ = vtkWebGPUFullScreenQuad.newInstance();
       model.clearFSQ.setDevice(device);
       model.clearFSQ.setPipelineHash('clearfsq');
-      model.clearFSQ.setFragmentShaderTemplate(clearFragTemplate);
+      model.clearFSQ.setFragmentShaderTemplate(clearFragColorTemplate);
       const ubo = vtkWebGPUUniformBuffer.newInstance({ label: 'mapperUBO' });
+      ubo.addEntry('FSQMatrix', 'mat4x4<f32>');
       ubo.addEntry('BackgroundColor', 'vec4<f32>');
       model.clearFSQ.setUBO(ubo);
+
+      model.backgroundTex = model.renderable.getEnvironmentTexture();
     }
+    // Textured Background
+    if (
+      model.clearFSQ.getPipelineHash() !== 'clearfsqwithtexture' &&
+      model.renderable.getUseEnvironmentTextureAsBackground() &&
+      model.backgroundTex?.getImageLoaded()
+    ) {
+      model.clearFSQ.setFragmentShaderTemplate(clearFragTextureTemplate);
+      const ubo = vtkWebGPUUniformBuffer.newInstance({ label: 'mapperUBO' });
+      ubo.addEntry('FSQMatrix', 'mat4x4<f32>');
+      ubo.addEntry('BackgroundColor', 'vec4<f32>');
+      model.clearFSQ.setUBO(ubo);
 
+      const environmentTextureHash = device
+        .getTextureManager()
+        .getTextureForVTKTexture(model.backgroundTex);
+      if (environmentTextureHash.getReady()) {
+        const tview = environmentTextureHash.createView(`EnvironmentTexture`);
+        model.clearFSQ.setTextureViews([tview]);
+        model.backgroundTexLoaded = true;
+        const interpolate = model.backgroundTex.getInterpolate()
+          ? 'linear'
+          : 'nearest';
+        tview.addSampler(device, {
+          addressModeU: 'repeat',
+          addressModeV: 'clamp-to-edge',
+          addressModeW: 'repeat',
+          minFilter: interpolate,
+          magFilter: interpolate,
+          mipmapFilter: 'linear',
+        });
+      }
+      model.clearFSQ.setPipelineHash('clearfsqwithtexture');
+    } else if (
+      model.clearFSQ.getPipelineHash() === 'clearfsqwithtexture' &&
+      !model.renderable.getUseEnvironmentTextureAsBackground()
+    ) {
+      // In case the mode is changed at runtime
+      model.clearFSQ = vtkWebGPUFullScreenQuad.newInstance();
+      model.clearFSQ.setDevice(device);
+      model.clearFSQ.setPipelineHash('clearfsq');
+      model.clearFSQ.setFragmentShaderTemplate(clearFragColorTemplate);
+      const ubo = vtkWebGPUUniformBuffer.newInstance({ label: 'mapperUBO' });
+      ubo.addEntry('FSQMatrix', 'mat4x4<f32>');
+      ubo.addEntry('BackgroundColor', 'vec4<f32>');
+      model.clearFSQ.setUBO(ubo);
+    }
+
+    const keyMats = model.webgpuCamera.getKeyMatrices(publicAPI);
     const background = model.renderable.getBackgroundByReference();
+
     model.clearFSQ.getUBO().setArray('BackgroundColor', background);
+    const fsqMatrix = [];
+    const transposedNormals = [];
```

Comment:
Let's allocate these two as statics so that we are not allocating and destroying them each frame. Ala 

const _fsqClearMat4 = new Float64Array(16);

Reply:
Should this be put somewhere that's only run once or is it fine to just directly swap [] with new Float64Array(16)? I'm not super familiar with javascript allocation

Reply:
Look at WebGPU\ImageMapper for an example